### PR TITLE
mapnik: Add patch to support building with boost v1.83

### PR DESCRIPTION
**Pull request by hummeltech**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/m/mapnik.rb [hunk 1]
```diff
@@ -17,6 +17,13 @@ class Mapnik < Formula
...
       sha256 "79a85ddba3ec17b86cb216e21442611498a9f2612f03e98708057b3c3a6e8b06"
     end
+
+    # Fix build with Boost v1.83 using Arch Linux patch. Remove in the next release.
+    # Ref: https://gitlab.archlinux.org/archlinux/packaging/packages/mapnik/-/commit/f9f723222c0af50e4db747e7c4e8138dbc769f53
+    patch do
+      url "https://gitlab.archlinux.org/archlinux/packaging/packages/mapnik/-/raw/f9f723222c0af50e4db747e7c4e8138dbc769f53/boost-1.83.patch"
+      sha256 "356271f4550c2b370ae48bbce9cebb58c5803507f2b14bc8e84f3813871d0645"
+    end
   end
```
[the remark this answers is not in the document]
Comment (hummeltech) [reply Formula/m/mapnik.rb:27]: This particular patch is indeed in use in the release `Arch Linux` package repository source. However, the upstream `mapnik` project does not appear to be actively working on any `3.x.x` releases and seems to be focusing on the upcoming `4.x.x` release (which has already addressed this issue).

[the remark this answers is not in the document]
Comment (hummeltech) [reply Formula/m/mapnik.rb:27]: As far as I can tell, there is no single commit that can be used as an alternative to this patch ([this](https://github.com/mapnik/mapnik/commit/50f1b05f863ab7060558d91a7d9521d849f29df3) is the only one I could find which specifically mentions Boost 1.83). Also, upon seeing the already included `Arch Linux` patch to support `SCONS 4`, I assumed using another would be acceptable. The same patch is also in use in the FreeBSD package.

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/m/mapnik.rb:27]: Can you still upstream it to Mapnik? If there ever is a repeat of https://www.debian.org/security/2008/dsa-1571 I'd like it not to be in Homebrew.

[the remark this answers is not in the document]
Comment (hummeltech) [reply Formula/m/mapnik.rb:27]: Alright, I can submit a pull request. Thanks

[the remark this answers is not in the document]
Comment (hummeltech) [reply Formula/m/mapnik.rb:27]: Alright, I have submitted a pull request, let me know if you have any questions. Thanks
https://github.com/mapnik/mapnik/pull/4413

[the remark this answers is not in the document]
Comment (hummeltech) [reply Formula/m/mapnik.rb]: Alright @SMillerDev, the patch has been merged and I have updated the patch URL to come from the upstream project.

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**